### PR TITLE
Implement 2 more UI Automated cases, take out AUTOMFA cases, minor tweaks to some cases

### PR DESCRIPTION
**Pull request by fadidurah**
UI Automation additions for December maintenance

- Add implementation for these two test cases:
(Mooncake) https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1592509
(Broker) https://identitydivision.visualstudio.com/Engineering/_testPlans/define?planId=2007357&suiteId=2008868
- AutoMFA cases are very inconsistent, taking those out of pipeline
- Minor tweaks to few test cases (add retries, add annotations to denote local brokerhost variant test cases)

Pre-req: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1921
Related: https://github.com/AzureAD/ad-accounts-for-android/pull/2092

### CODE REVIEW

**Review by melissaahn — APPROVED** [review]

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java [hunk 1]
```diff
@@ -38,6 +39,7 @@
 // Flight settings
 // https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1561087
 @SupportedBrokers(brokers = BrokerHost.class)
+@LocalBrokerHostDebugUiTest
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java:42]: Isn't supported brokers annotation doing similar thing?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561136.java [hunk 2]
```diff
@@ -106,12 +109,14 @@ public void handleUserInteraction() {
...
-        return null;
+        return LabQuery.builder()
+                .azureEnvironment(AzureEnvironment.AZURE_CLOUD)
+                .build();
     }
 
     @Override
     public TempUserType getTempUserType() {
-        return TempUserType.BASIC;
+        return null;
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561136.java:119]: Why changing this?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase497044.java [hunk 3]
```diff
@@ -45,8 +46,9 @@
 
 // Interactive auth w/ force_login w/ MFA
 // https://identitydivision.visualstudio.com/DefaultCollection/DevEx/_workitems/edit/497044
-@RetryOnFailure(retryCount = 3) // Seems like AutoMFA does not work sometimes, seems rare but adding extra retries
+@RetryOnFailure(retryCount = 2)
 @RunOnAPI29Minus("Verify Your Identity (MFA)")
+@DoNotRunOnPipeline("Looks like AutoMfa From lab account is still very inconsistent, don't run this on pipeline")
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase497044.java:51]: Have we checked with Lab team about this?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java (hunk 1 above)
Comment (fadidurah) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java:42]: Yes, the issue is that from what I researched you can't use @SupportedBrokers with a particular value for "brokers" when assigning test targets (i.e. you can't assign a test run to ignore tests that have the SupportedBrokers annotation with value BrokerHost.class). To be able to create this test target distinction, we need an annotation specifically for tests that requires msal test app built with Local BrokerHost Debug, hence adding this new annotation

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561136.java (hunk 2 above)
Comment (fadidurah) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561136.java:119]: Saves some time, as there is a wait after creating a temp user. There doesn't seem to be a strong distinction that the test requires creation of a new Temporary user (I see it says "create", but I believe we've interpreted that in the past as meaning either create a temp user or fetch an existing basic account). I can change it back if we feel this is affecting the integrity of the test.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase497044.java (hunk 3 above)
Comment (fadidurah) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase497044.java:51]: I have reached out to Ryan already, but I think he is oof for holidays. Will have to wait until he comes back

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**